raster-foundry/raster-foundry
# Copy failed tifs to debug data bucket

## Overview

This PR uploads tifs that fail to process to a debug data bucket so we can figure out what went wrong even if
people sensibly follow our advice about deleting their failed projects.

### Checklist


## Testing Instructions

- raise an exception (any exception) at the beginning of the `GeoTiffS3SceneFactory`
- rebuild the batch container -- `docker-compose build batch`
- if you haven't in a while, rebuild the batch jar from sbt: `batch/assembly`
- fire up groundwork and your RF servers and create an upload with the network tab open
- copy your upload's id
- try to process your upload, pretending that this is the third job attempt: `AWS_BATCH_JOB_ATTEMPT=3 ./scripts/console batch 'rf process-upload d64e5be4-bf8c-4c82-90bd-d7ea9ad4082d'`
- you should see a message beginning `Skipping debug data upload`
- set a debug data bucket that you can write to (any bucket will do but I have a suggestion)
- run the upload processing command again
- You should see a message ending `It will be available in the debug bucket for 14 days.`
- check the bucket / key that that message reported -- there should be a tif there
- confirm that the environment variable for the debug data bucket matches what's in azavea/raster-foundry-deployment#255

Closes #1049
